copier.go

```diff
@@ -66,5 +69,92 @@ func TestCopier(t *testing.T, store types.Storager) {
 				})
 			})
 		})
+
+		Convey("When Copy to an existing file", func() {
+			srcSize := rand.Int63n(4 * 1024 * 1024) // Max file size is 4MB
+			content, _ := ioutil.ReadAll(io.LimitReader(randbytes.NewRand(), srcSize))
+			src := uuid.New().String()
+
+			_, err := store.Write(src, bytes.NewReader(content), srcSize)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			dstSize := rand.Int63n(4 * 1024 * 1024) // Max file size is 4MB
+			r := io.LimitReader(randbytes.NewRand(), dstSize)
+			dst := uuid.New().String()
+
+			_, err = store.Write(dst, r, dstSize)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			defer func() {
+				err = store.Delete(src)
+				if err != nil {
+					t.Error(err)
+				}
+				err = store.Delete(dst)
+				if err != nil {
+					t.Error(err)
+				}
+			}()
+
+			err = c.Copy(src, dst)
+			Convey("The error should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+
+			Convey("Read should get dst object data without error", func() {
+				var buf bytes.Buffer
+				n, err := store.Read(dst, &buf)
+
+				Convey("The error should be nil", func() {
+					So(err, ShouldBeNil)
+				})
+
+				Convey("The content should be match", func() {
+					So(buf, ShouldNotBeNil)
+					So(n, ShouldEqual, srcSize)
+					So(md5.Sum(buf.Bytes()), ShouldResemble, md5.Sum(content))
+				})
+			})
+		})
+
+		Convey("When Copy to an existing dir", func() {
+			d, ok := store.(types.Direr)
```

Comment:
Service that didn't support `CreateDir` could be failed, we can check and skip here if not supported.